core/base-service/legacy-request-handler.spec.js

```diff
@@ -1,251 +0,0 @@
-      message: someValue,
-    },
-    {}
-  )
-  sendBadge(format, badgeData)
-}
-
-function createFakeHandlerWithCacheLength(cacheLengthSeconds) {
-  return function fakeHandler(queryParams, match, sendBadge, request) {
-    const [, someValue, format] = match
-    const badgeData = coalesceBadge(
-      queryParams,
-      {
-        label: 'testing',
-        message: someValue,
-      },
-      {},
-      {
-        _cacheLength: cacheLengthSeconds,
-      }
-    )
-    sendBadge(format, badgeData)
-  }
-}
-
-describe('The request handler', function () {
-  let port, baseUrl
-  beforeEach(async function () {
-    port = await portfinder.getPortPromise()
-    baseUrl = `http://127.0.0.1:${port}`
-  })
-
-  let camp
-  beforeEach(function (done) {
-    camp = Camp.start({ port, hostname: '::' })
-    camp.on('listening', () => done())
-  })
-  afterEach(function (done) {
-    if (camp) {
-      camp.close(() => done())
-      camp = null
-    }
-  })
-
-  const standardCacheHeaders = { defaultCacheLengthSeconds: 120 }
-
-  describe('the options object calling style', function () {
-    beforeEach(function () {
-      camp.route(
-        /^\/testing\/([^/]+)\.(svg|png|gif|jpg|json)$/,
-        handleRequest(standardCacheHeaders, { handler: fakeHandler })
-      )
-    })
-
-    it('should return the expected response', async function () {
-      const { statusCode, body } = await got(`${baseUrl}/testing/123.json`, {
-        responseType: 'json',
-      })
-      expect(statusCode).to.equal(200)
-      expect(body).to.deep.equal({
-        name: 'testing',
-        value: '123',
-        label: 'testing',
-        message: '123',
-        color: 'lightgrey',
-        link: [],
-      })
-    })
-  })
-
-  describe('the function shorthand calling style', function () {
-    beforeEach(function () {
-      camp.route(
-        /^\/testing\/([^/]+)\.(svg|png|gif|jpg|json)$/,
-        handleRequest(standardCacheHeaders, fakeHandler)
-      )
-    })
-
-    it('should return the expected response', async function () {
-      const { statusCode, body } = await got(`${baseUrl}/testing/123.json`, {
-        responseType: 'json',
-      })
-      expect(statusCode).to.equal(200)
-      expect(body).to.deep.equal({
-        name: 'testing',
-        value: '123',
-        label: 'testing',
-        message: '123',
-        color: 'lightgrey',
-        link: [],
-      })
-    })
-  })
-
-  describe('caching', function () {
-    describe('standard query parameters', function () {
-      function register({ cacheHeaderConfig }) {
-        camp.route(
-          /^\/testing\/([^/]+)\.(svg|png|gif|jpg|json)$/,
-          handleRequest(
-            cacheHeaderConfig,
-            (queryParams, match, sendBadge, request) => {
-              fakeHandler(queryParams, match, sendBadge, request)
-            }
-          )
-        )
-      }
-
-      it('should set the expires header to current time + defaultCacheLengthSeconds', async function () {
```

Comment:
As I mentioned in the top comment it may make sense to comb through these behaviors and see if we should add test coverage elsewhere.